src/libraries/System.Private.CoreLib/src/System/DefaultBinder.cs

```diff
@@ -196,43 +196,54 @@ internal partial class DefaultBinder : Binder
                     if (pCls.IsByRef)
                         pCls = pCls.GetElementType()!;
 
-                    // the type is the same
-                    if (pCls == argTypes[paramOrder[i][j]])
-                        continue;
+                    int index = paramOrder[i][j];
+                    if (index < args.Length)
```

Comment:
In case only 1st and 3rd named arguments provided from the 4 parameters the order in `paramOrder[i]` will be {0, 2, 1, 3} and `argTypes[paramOrder[i][1]]` will cause `IndexOutOfBoundsException`, therefore first need to check if current parameter is within provided arguments range